### PR TITLE
Vulkancore

### PR DESCRIPTION
**Pull request by Qozul**
Changes include handling of different colour attachment counts, drawing depth buffer when fragment shader is removed, a folder restructure, and better code in descriptors allow for built in (system) descriptors. 

Validation against physical device limits of descriptors and config has been added. User accessible validation has also been added, currently with limited functionality.

### CODE REVIEW

File: Resources/Shaders/Source/depth_frag.frag [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#version 450
+
+layout(location = 0) in vec2 inTexcoord;
+
+layout(location = 0) out vec4 outColour;
+
+layout(set = 0, binding = 0) uniform sampler2D depthTexture;
+
+float LinearizeDepth(float depth, float near, float far)
+{
```
Comment (Kney-Delach) [inline Resources/Shaders/Source/depth_frag.frag]: Change to same line bracet to maintain consistency with codebbase.

File: Resources/Shaders/Source/depth_frag.frag [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	return (2.0 * near) / (far + near - depth * (far - near));
+}
+
+void main() 
+{
```
Comment (Kney-Delach) [inline Resources/Shaders/Source/depth_frag.frag]: Same as above.

File: Resources/Shaders/Source/fullscreen.vert [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#version 450
+
+layout(location = 0) out vec2 outTexcoord;
+
+out gl_PerVertex {
+	vec4 gl_Position;
+};
+
+void main() 
+{
```
Comment (Kney-Delach) [inline Resources/Shaders/Source/fullscreen.vert]: Change to same line bracket.

**Review by Kney-Delach — APPROVED** [review]
View comments before merging. Other than these notes, you are safe to merge.

File: Resources/Shaders/Source/fullscreen.vert [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	outTexcoord = vec2((gl_VertexIndex << 1) & 2, gl_VertexIndex & 2);
+    gl_Position = vec4(outTexcoord * 2.0 - 1.0, 0.0, 1.0);
```
Comment (SzyJ) [inline Resources/Shaders/Source/fullscreen.vert:14]: Spaces and tabs used here on different lines. Please make consistent. Also { bracket is on same line in all other code

Comment (Qozul) [reply Resources/Shaders/Source/fullscreen.vert:14]: Should be fixed now to use spaces consistently in the file.

**Pull request merged**